olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/TileRequest.h

```diff
@@ -0,0 +1,212 @@
+    tile_key_ = std::move(tile_key);
+    return *this;
+  }
+
+  /**
+   * @brief Sets the tile key for the request.
+   *
+   * @param tile_key Tile key value.
+   *
+   * @return A reference to the updated `TileRequest` instance.
+   */
+  inline TileRequest& WithTileKey(const geo::TileKey& tile_key) {
+    tile_key_ = tile_key;
+    return *this;
+  }
+
+  /**
+   * @brief Gets a tile key value.
+   *
+   * @return The tile key or `boost::none` if tile_key is not set.
+   */
+  inline const boost::optional<geo::TileKey>& GetTileKey() const {
+    return tile_key_;
+  }
+
+  /**
+   * @brief Sets offset value for the request.
+   *
+   * @param offset Offset value, it must be from 0 to 4. By default offset is 2.
+   *
+   * @return A reference to the updated `TileRequest` instance.
+   */
+  inline TileRequest& WithOffset(unsigned int offset) {
+    offset_ = offset;
+    return *this;
+  }
+
+  /**
+   * @brief Gets the offset.
+   *
+   * @return offset value.
+   */
+  inline unsigned int GetOffset() const { return offset_; }
+
+  /**
+   * @brief Sets depth value for the request.
+   *
+   * @param depth Depth value. The maximum allowed value for the depth parameter
+   * is 4. By default depth is 4.
+   *
+   * @return A reference to the updated `TileRequest` instance.
+   */
+  inline TileRequest& WithDepth(unsigned int depth) {
+    depth_ = depth;
+    return *this;
+  }
+
+  /**
+   * @brief Gets a depth value.
+   *
+   * @return depth value.
+   */
+  inline unsigned int GetDepth() const { return depth_; }
+
+  /**
+   * @brief Gets the fetch option that controls how requests are handled.
+   *
+   * The default option is `OnlineIfNotFound` that queries the network if
+   * the requested resource is not in the cache.
+   *
+   * @return The fetch option.
+   */
+  inline FetchOptions GetFetchOption() const { return fetch_option_; }
+
+  /**
+   * @brief Sets the fetch option that you can use to set the source from
+   * which data should be fetched.
+   *
+   * @see `GetFetchOption()` for information on usage and format.
+   *
+   * @param fetch_option The `FetchOption` enum.
+   *
+   * @return A reference to the updated `TileRequest` instance.
+   */
+  inline TileRequest& WithFetchOption(FetchOptions fetch_option) {
+    fetch_option_ = fetch_option;
+    return *this;
+  }
+
+  /**
+   * @brief Creates a readable format for the request.
+   *
+   * @return A string representation of the request.
+   */
+  inline std::string CreateKey() const {
+    std::stringstream out;
+    if (GetTileKey()) {
+      out << "key:" << GetTileKey().get();
+    }
+    if (GetBillingTag()) {
+      out << "$" << GetBillingTag().get();
+    }
+    out << "^" << GetFetchOption();
+    out << "[";
+    out << GetOffset() << ":" << GetDepth();
+    out << "]";
+    return out.str();
+  }
+
+ private:
+  boost::optional<std::string> billing_tag_;
+  boost::optional<geo::TileKey> tile_key_;
+  unsigned int offset_{2};
+  unsigned int depth_{4};
```

Comment:
I dont think we should add this for now. As discussed with Mischa these we can do later if users request it to optimize download. But for now, I don't see any reason why we should allow.

Reply:
Deleted for now